### ISSUE TITLE
Differential backup fails because cached item path ends up in list of srcs to upload

### ISSUE THREAD

**Issue by srjacobs** [thelastpickle/cassandra-medusa#108]
https://github.com/thelastpickle/cassandra-medusa/blob/8805a163256a2dff982b4cdd1bc3709412dbf2ca/medusa/backup.py#L152

When doing a differential backup and the last backup was Full then you get to this point in the code where it puts the cached item path into the retained list.
The caller of  replace_or_remove_if_cached uses the retained list as the list of srcs for files that need to be backed up.  The medusa backup fails with an error at this line https://github.com/thelastpickle/cassandra-medusa/blob/8805a163256a2dff982b4cdd1bc3709412dbf2ca/medusa/storage/aws_s3_storage/concurrent.py#L103  :

ERROR: This error happened during the backup: [Errno 2] No such file or directory: 'staging/......

and staging is the prefix defined for my s3 bucket.

**Comment by rzvoncek** [comment]
Hi @srjacobs , and thanks for the report. I was able to reproduce this easily by adding a scenario doing a full backup followed by a differential one against AWS.

The fix, in short, is to make the 1st differential backup upload everything, even if a full backup exists. We need to do this because both backup modes use a different file structure in the storage backend.

I've opened https://github.com/thelastpickle/cassandra-medusa/pull/135 for the change.
